draco/analysis/transform.py

```diff
@@ -226,13 +225,19 @@ def process(self, ss):
             **output_kwargs,
         )
 
+        # Check if frequencies are already ordered
+        no_redistribute = freq_ind == sorted(freq_ind)
```

Comment:
This test is not sufficient I think as `freq_ind` may not represent all the indices in the input array (i.e. it may be down selecting), but the selection may already be sorted.

At the least this test needs changing, probably to something like:
```suggestion
        no_redistribute = freq_ind == list(range(len(ss.freq)))
```

However, I wonder if it would be better to just remove the frequency down selection code and put into a separate task? One that takes a `Telescope` object as a requirement and then just does the same frequency selection that's in this task?

Reply:
If I'm understanding correctly, you mean we would have one task that downselects the frequencies followed by `CollateProducts` which would re-order them (if needed)?